### DISCUSSION TITLE
How to trigger a dag refresh via API?

### DISCUSSION THREAD

**Discussion by jakubgs** [apache/airflow#15031] [Q&A] [answered]
I've written a simple automation that updates my DAGs based on repo pushes using GitHub webhooks. My issue is that the refresh interval makes it sometimes annoyingly slow to pick up changes.

Is there a way to make Airflow refresh DAGs using something like an API call?

I looked at the [API documentation](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#tag/DAG) but I don't see anything like `POST /dags/{dag_id}/refresh`. Or am I blind?

**Comment by turbaszek** [comment]
What do you mean by refreshing DAG?

**Reply by jakubgs** [reply]
I mean, the folder that contains Python DAG definitions is scanned every N seconds. Is there a way to trigger a scan of that folder via API?

**Comment by XD-DENG** [comment, marked as answer]
This is handled separately by `DagFileProcessorManager` (broadly speaking, under `Scheduler`'s scope). Web API doesn't provide any interface for the use case you mentioned ("*trigger a scan of DAG folder*").

So the short answer is: currently there is no such a way.

**Reply by turbaszek** [reply]
XD is right. Also I don't see reason why we would need such function - the DAGs should be slowly changing so there's no need to click "update right now". Moreover, the speed of dag parser in 2.0 seems to be vastly increased when compared to 1.10
